### PR TITLE
[imgui] Update to 1.91.0

### LINKED ISSUE #40178: [imgui] update to 1.91.0
### Library name

imgui

### New version number

1.91.0

### Other information that may be useful (release notes, etc...)

https://github.com/ocornut/imgui/releases/tag/v1.91.0

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/40178
Added conversion of multibyte strings to wide strings.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

All features passed with following triplets:

```
x64-windows
x64-windows-static
```
Usage test pass with following triplets:

```
x64-windows
```

### CODE REVIEW

File: ports/hello-imgui/fix-upw.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+diff --git a/src/hello_imgui/internal/poor_man_log.cpp b/src/hello_imgui/internal/poor_man_log.cpp
+index 3aad6f7..2fb93fc 100644
+--- a/src/hello_imgui/internal/poor_man_log.cpp
++++ b/src/hello_imgui/internal/poor_man_log.cpp
+@@ -18,7 +18,13 @@ void PoorManLog(const char* msg, ...)
+     va_end(args);
+
+ #ifdef _MSC_VER
++#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
```
Comment (BillyONeal) [inline ports/hello-imgui/fix-upw.patch:9]: This is *probably* OK but not sure. I think upstream needs a look.

Comment (jimwang118) [reply ports/hello-imgui/fix-upw.patch:9]: Ok, I'll report this change upstream.

**Comment by pthom** [comment]
Hi, thanks a lot for having a look at this!

I think there is a small typo in this PR. Here is the code after applying it: `OutputDebugString` is called twice.

```cpp
#ifdef _MSC_VER
#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
    wchar_t wbuffer[2000];
    mbstowcs(wbuffer, buffer, strlen(buffer) + 1);
    OutputDebugString(wbuffer);
    OutputDebugString(buffer);
#else
     OutputDebugString(buffer);
#endif
#else
    printf("%s", buffer);
#endif
```

Anyhow, this function is very rarely used, for internal debugging purposes. It will always receive latin characters.
So, if the code below compiles correctly with UWP, I think it might be enough and easier to maintain:

```cpp
#ifdef _MSC_VER
     OutputDebugStringA(buffer);   // Force select ANSI version (for uwp compatibility)
#else
    printf("%s", buffer);
#endif
```

If you have time, could you please test this version with vcpkg?


Thanks a lot!

**Comment by pthom** [comment]
Since the new patch seems to pass uwp in the CLI here, I pushed it [upstream](https://github.com/pthom/hello_imgui/commit/5ebc73633932e30bf2426d378657ca820c659d3a) inside hello_imgui

Thanks!

**Review by JonLiu1993 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**